.github/workflows/ci.yaml

```diff
@@ -692,43 +692,64 @@ jobs:
           go mod download
 
           version="$(./scripts/version.sh)"
+          tag="main-$(echo "$version" | sed 's/+/-/g')"
+          echo "tag=$tag" >> $GITHUB_OUTPUT
+
           make gen/mark-fresh
           make -j \
-            build/coder_linux_amd64 \
+            build/coder_linux_{amd64,arm64,armv7} \
             build/coder_"$version"_windows_amd64.zip \
             build/coder_"$version"_linux_amd64.{tar.gz,deb}
 
-      - name: Build and Push Linux amd64 Docker Image
+      - name: Build Linux Docker images
         id: build-docker
+        env:
+          CODER_IMAGE_BASE: ghcr.io/coder/coder-preview
+          DOCKER_CLI_EXPERIMENTAL: "enabled"
         run: |
           set -euxo pipefail
+
+          # build Docker images for each architecture
           version="$(./scripts/version.sh)"
           tag="main-$(echo "$version" | sed 's/+/-/g')"
-
-          export CODER_IMAGE_BUILD_BASE_TAG="$(CODER_IMAGE_BASE=coder-base ./scripts/image_tag.sh --version "$version")"
-          ./scripts/build_docker.sh \
-            --arch amd64 \
-            --target "ghcr.io/coder/coder-preview:$tag" \
-            --version $version \
-            --push \
-            build/coder_linux_amd64
-
-          # Tag as main
-          docker tag "ghcr.io/coder/coder-preview:$tag" ghcr.io/coder/coder-preview:main
-          docker push ghcr.io/coder/coder-preview:main
-
-          # Store the tag in an output variable so we can use it in other jobs
           echo "tag=$tag" >> $GITHUB_OUTPUT
 
+          # build images for each architecture
+          make -j build/coder_"$version"_linux_{amd64,arm64,armv7}.tag
+
+          # only push if we are on main branch
+          if [ "${{ github.ref }}" == "refs/heads/main" ]; then
+            # build and push multi-arch manifest, this depends on the other images
+            # being pushed so will automatically push them
+            make -j push/build/coder_"$version"_linux_{amd64,arm64,armv7}.tag
+
+            # Create and push a multi-arch manifest
+            ./scripts/build_docker_multiarch.sh \
+              --push \
+              --target "ghcr.io/coder/coder-preview:$tag" \
+              --version $version \
+              $(cat build/coder_"$version"_linux_{amd64,arm64,armv7}.tag)
+
+            # Create and push a multi-arch manifest for the latest tag
+            ./scripts/build_docker_multiarch.sh \
+              --push \
+              --target "ghcr.io/coder/coder-preview:latest" \
```

Comment:
Can we keep the tag at `:main`?

Reply:
Yes, but I think as the image name is `coder-preview,` choosing `main` as the tag was not a good decision at the start. Thastw hy I am renaming it to `latest`
docker fulls to pull an image that does not have a tag in the name if there is no `latest` tag.

Reply:
I am using `ghcr.io/coder/coder-preview:main` to pull the latest image every 5 minutes using `watchtower` and as per k8s [docs](https://kubernetes.io/docs/concepts/containers/images/#imagepullpolicy-defaulting),

> [!NOTE]
> - if you omit the `imagePullPolicy` field, and the tag for the container image is `:latest`, `imagePullPolicy` is automatically set to `Always`.
> - if you omit the `imagePullPolicy` field, and you specify the tag for the container image that isn't `:latest`, the `imagePullPolicy` is automatically set to `IfNotPresent`.

Reply:
We are keeping `main` and adding `latest` as an additional tag.